src/processors/file-handler.js

```diff
@@ -103,6 +103,7 @@ export async function* readBatch(asyncIterator, fileName) {
   for await (const batch of asyncIterator) {
     // Last batch will have this special type and will provide all the root
     // properties of the parsed document.
+    // Only json parse will have `FINAL_RESULT`
```

Comment:
We should probably not rely on this as a way of detecting json parsing, in the future some other loader might also emit a `FINAL_RESULT` batch.

But the check for jsonpath should be robust enough.